django/core/checks/urls.py

```diff
@@ -14,17 +14,19 @@ def check_resolver(resolver):
     """
     Recursively check the resolver.
     """
-    from django.core.urlresolvers import RegexURLPattern, RegexURLResolver
+    from django.urls import Resolver, ResolverEndpoint
     warnings = []
-    for pattern in resolver.url_patterns:
-        if isinstance(pattern, RegexURLResolver):
-            warnings.extend(check_include_trailing_dollar(pattern))
-            # Check resolver recursively
-            warnings.extend(check_resolver(pattern))
-        elif isinstance(pattern, RegexURLPattern):
-            warnings.extend(check_pattern_name(pattern))
+    # We only check when we're using a built-in resolver
+    if isinstance(resolver, Resolver):
+        for resolver in resolver.resolvers:
```

Comment:
I would avoid cloaking the `resolver` variable, I like your `sub_resolver` naming above.